File: src/main/java/com/rs/game/world/entity/player/DialogueManager.java
package com.rs.game.world.entity.player;

import com.rs.game.world.entity.player.content.dialogue.Dialogue;
import com.rs.game.world.entity.player.content.dialogue.DialogueHandler;
import com.rs.utility.Logger;

public class DialogueManager {

	private Player player;
	private Dialogue lastDialogue;


	public DialogueManager(Player player) {
		this.player = player;
	}

	public void startDialogue(Object key, Object... parameters) {
		if (!player.getControlerManager().useDialogueScript(key))
			return;
		if (lastDialogue != null)
			lastDialogue.finish();
		lastDialogue = DialogueHandler.getDialogue(key);
		if (lastDialogue == null)
			return;
		lastDialogue.parameters = parameters;
		lastDialogue.setPlayer(player);
		lastDialogue.start();
		Logger.logControler("["+player.getUsername()+"] has used the dialogue "+key+".");
	}

	public boolean hasDialogue() {
		return lastDialogue !=null;
	}

	public void continueDialogue(int interfaceId, int componentId) {
		if (interfaceId == 13 && componentId == 6
				|| interfaceId == 13 && componentId == 7
				|| interfaceId == 13 && componentId == 8
				|| interfaceId == 13 && componentId == 9
				|| interfaceId == 13 && componentId == 10
				|| interfaceId == 13 && componentId == 11
				|| interfaceId == 13 && componentId == 12
				|| interfaceId == 13 && componentId == 13
				|| interfaceId == 13 && componentId == 14
				|| interfaceId == 13 && componentId == 15) {
			player.getBankPin().handleButtons(interfaceId, componentId);
			return;
		}
		if (interfaceId == 14 && componentId == 18
				|| interfaceId == 14 && componentId == 33
				|| interfaceId == 14 && componentId == 35
				|| interfaceId == 14 && componentId == 19) {
			player.getBankPin().handleButtons(interfaceId, componentId);
			return;
		}
		if (lastDialogue == null)
			return;
		lastDialogue.run(interfaceId, componentId);
	}

	public void finishDialogue() {
		if (lastDialogue == null)
			return;
		lastDialogue.finish();
		lastDialogue = null;
		if (player.getInterfaceManager().containsChatBoxInter())
			player.getInterfaceManager().closeChatBoxInterface();
	}

}

File: src/main/kotlin/com/hyze/plugins/dialogue/DialogueManager.kt
package com.hyze.plugins.dialogue

import com.hyze.plugins.dialogue.message.OptionMessageDialogue
import com.rs.game.world.entity.player.Player

class DialogueManager(val player: Player) {

    private var lastDialogue: DialogueBuilder? = null

    /**
     * Sends player to the next dialogue stage
     */

    fun next(interfaceId: Int, componentId: Int) {
        if (interfaceId == 1184) {
            lastDialogue?.callNextMessage()
        } else if (interfaceId == 1188) {
            lastDialogue?.callNextMessage(OptionMessageDialogue.Option.getOption(componentId))
        }
    }

    /**
     * Start a player dialogue
     */

    fun start(dialogueBuilder: DialoguePlugin, npcId: Int) {
        if (lastDialogue != null) {
            finish()
        }

        val plugin = dialogueBuilder.build(player, npcId)
        lastDialogue = plugin

        plugin.callMessage()
    }

    /**
     * Finish a dialogue
     */

    fun finish() {
        if (lastDialogue == null) return

        lastDialogue!!.end.invoke()

        lastDialogue = null
        if (player.interfaceManager.containsChatBoxInter()) {
            player.interfaceManager.closeChatBoxInterface()
        }
    }

}